### ISSUE TITLE
MangoHud flatpak end-of-life

### ISSUE THREAD

**Issue by juligreen** [flathub/net.lutris.Lutris#175]
When updating today, I got the following message:

```
com.valvesoftware.Steam.Utility.MangoHud//stable is end-of-life, with reason:
   This extension was converted to a runtime extension
```

Should there be another way to install mangohud into lutris then?

**Comment by A6GibKm** [comment]
Yes it was replaced by org.freedesktop.Platform.VulkanLayer.MangoHud. I will check it during the week.

**Comment by A6GibKm** [comment]
As a headsup, lutris was updated to the GNOME 41 runtime, now it should be possible to use the extension, I am just not entirely sure how.

**Comment by falk4243** [comment]
That is a good question: org.freedesktop.Platform.VulkanLayer.MangoHud does not seem to be recognized by Lutris ... and, strange enough, neither by Steam on my machine (Debian 11, with flatpak 1.11.3-2) for that matter. The latter still recognizes the old Mangohud utility, which was also the case for Lutris until the latest update. Now the "FPS counter" box is just greyed out, no matter which package is installed. Any input would be appreciated!

**Comment by rowbawts** [comment]
> That is a good question: org.freedesktop.Platform.VulkanLayer.MangoHud does not seem to be recognized by Lutris ... and, strange enough, neither by the most recent Steam flatpak (stable) on my machine (Debian 11, with flatpak 1.11.3-2 from testing) for that matter. The latter still recognizes the old Mangohud utility, which was also the case for Lutris until the latest update. Now the "FPS counter" box is just greyed out, no matter which package is installed. Any input would be appreciated!

If you use this as your launch options on Steam the new Flatpak will work but I tried the same on Lutris and got nothing

`/usr/lib/extensions/vulkan/MangoHud/bin/mangohud %command%`

**Comment by A6GibKm** [comment]
There is a WIP patch for Steam, we might want to take  a similar approach so that it is just plug n play.

cc @gasinvein

**Comment by gasinvein** [comment]
GNOME runtime didn't (forget to?) add the `.VulkanLayer` extension point, see [here](https://gitlab.gnome.org/GNOME/gnome-build-meta/-/issues/426).

**Comment by falk4243** [comment]
> 
> If you use this as your launch options on Steam the new Flatpak will work but I tried the same on Lutris and got nothing
> 
> `/usr/lib/extensions/vulkan/MangoHud/bin/mangohud %command%`

Alternative (that doesn't work for Lutris yet, see above): Add the following env variable to your Steam flatpak: XDG_DATA_DIRS=/app/share:/usr/lib/extensions/vulkan/share:/usr/share:/usr/share/runtime/share:/run/host/user-share:/run/host/share

Afterward, you can control mangohud with MANGOHUD=1 and MANGOHUD_CONFIGFILE=PATH as usual, either on a per game basis or for the flatpak in general.

**Comment by gasinvein** [comment]
@falk4243 Your comment is misleading. Runtime's MangoHud doesn't work for Lutris because GNOME Runtime doesn't have the extension point. The `XDG_DATA_DIRS` trick is needed only for official Proton in Stream, it's irrelevant here.

**Comment by falk4243** [comment]
You're right, sorry for that - I have just edited my comment accordingly.

**Comment by A6GibKm** [comment]
The issue was resolved upstream, but due to how lutris detects mango hud, an MR had to be made so it is in the PATH https://github.com/flathub/net.lutris.Lutris/pull/188.

**Comment by A6GibKm** [comment]
Merged. Please test it and re open the issue if there is trouble.
